Changes based on Yieldable interface and Implementation of the NextDay feature

angelaaajing opened the pull request:
1. Change the return type of harvest method in plot class and getYield method in product class from String to Product type
2. Remove methods getYield and getDaysToYield from Product and add them to Interface Yieldable
3. Make Interface Yieldable be implmented by classes like Cow and Chicken, but not by Milk, Egg, etc.
4. Remove attributes yield and daysToYield and directly add info to methods getYield and getDaysToYield
5. Partially implement NextDay class and create NextDayController

angelaaajing committed:
1. Change the return type of harvest method in plot class and getYield method in product class from String to Product type
2. Remove methods getYield and getDaysToYield from Product and add them to Interface Yieldable
3. Make Interface Yieldable be implmented by classes like Cow and Chicken, but not by Milk, Egg, etc.
4. Remove attributes yield and daysToYield and directly add info to methods getYield and getDaysToYield
5. Partially implement NextDay class and create NextDayController

angelaaajing committed:
Revert "1. Change the return type of harvest method in plot class and getYield method in product class from String to Product type"

This reverts commit 67f1b341db666b05dc69ff8953a9c0b0b5d3fd91.

angelaaajing committed:
1. Change the return type of harvest method in plot class and getYield method in product class from String to Product type
2. Remove methods getYield and getDaysToYield from Product and add them to Interface Yieldable
3. Make Interface Yieldable be implmented by classes like Cow and Chicken, but not by Milk, Egg, etc.
4. Remove attributes yield and daysToYield and directly add info to methods getYield and getDaysToYield
5. Partially implement NextDay class and create NextDayController

angelaaajing committed:
Update Potato.java

angelaaajing committed:
Update Tomato.java

DavidGorospe reviewed (dismissed):
- Renaming of the entities packages and the use_case_interactors package abide by the package naming conventions
- CornCob shouldn't be a Yieldable because it doesn't yield anything so the change to the CornCob subclass to not implement the interface is correct. This is the same for Eggs, and Wool.
- Java documentation is accurate and informing
- NextDayController abides by dependency rule and correctly calls the necessary functions to update the state of the game
- nextDayInventory method correctly updates money by calling the removeRent method in InventoryManager by subtracting rent from user's money
- Might have to edit the feature  when crops/animals are ready to yield, they get auto-yielded. We might change this to not be auto-yielded for increased difficulty
- nextDayPlots correctly updates each plots' growth/yield progress
- nextDayEndgame correctly checks the win/lose conditions at the end of the day

angelaaajing committed:
Update Yieldable.java

angelaaajing committed:
Revert "1. Change the return type of harvest method in plot class and getYield method in product class from String to Product type"

This reverts commit 6ed89e09f68899dd2a5fac48911e40fb14cdf86a.

angelaaajing committed:
Merge branch '10-feature-9-go-to-next-day' of https://github.com/CSC207-2022F-UofT/course-project-jds-110 into 10-feature-9-go-to-next-day

angelaaajing committed:
1. Partially implement NextDay class, Controller and test for next day feature
2. Change the return type of getYield method in Product from String to Product type
3. Change the return type of harvest method in Plot from String to Product type

angelaaajing committed:
Revert "1. Partially implement NextDay class, Controller and test for next day feature"

This reverts commit c7d62d59cc15e553bf2a38d03d8b41bdf7643f6f.

dianzrong reviewed (approved):
Good job renaming packages, changing parts of the entities and implementing NextDay stuff!

Note: in NextDay, when yields are added to inventory if products are ripe, is the plot then made empty?
